### ISSUE TITLE
JSX parsing error on .tsx

### ISSUE THREAD

**Issue by mizchi** [eslint/typescript-eslint-parser#376]
**What version of TypeScript are you using?**

2.4

**What version of `typescript-eslint-parser` are you using?**

7.0.0

**What code were you trying to parse?**

```js
import * as React from 'react'
type Props = {
  title: string
}

export default function App(props: Props) {
  return (
    <h1>
      {props.title}
    </h1>
  )
}
```

**What did you expect to happen?**

Pass jsx syntax correctly.

**What happened?**

```
/Users/mz/sandbox/ts-proj/src/components/App.tsx
  9:12  error  Parsing error: ':' expected
```

**Comment by JamesHenry** [comment]
Thanks so much for your patience on this @mizchi!

This highlighted to me the fact that we currently have no way of testing TSX files specifically. We have typescript-specific construct tests, and JSX tests, but no way to explicitly test `.tsx` files.

I am addressing that in a PR and have used the source you have provided as the first test case. It seems that your issue has been resolved at some point, either in this project or in TypeScript itself, as the source no longer produces an error on latest master of this project and its supported stable TypeScript version of 2.5.x.

If you still have the issue when using the latest supported version please feel free to repoen, but otherwise I will be closing this when my PR lands.

Thanks again!
